lib/wallet/path.js

```diff
@@ -233,18 +243,46 @@ class Path extends bio.Struct {
 
   /**
    * Convert path to a json-friendly object.
+   * @param {String|Network?} network - Network type.
    * @returns {Object}
    */
 
-  getJSON() {
+  getJSON(network) {
     return {
       name: this.name,
       account: this.account,
       change: this.branch === 1,
-      derivation: this.toPath()
+      derivation: this.toPath(network)
     };
   }
 
+  /**
+   * Inject properties from a json object.
+   * @param {Object} json
+   */
+
+  fromJSON(json) {
+    assert(json && typeof json === 'object');
+    assert(json.derivation && typeof json.derivation === 'string');
+
+    const path = json.derivation.split('/');
+    const index = parseInt(path.pop(), 10);
+    const branch = parseInt(path.pop(), 10);
+    const account = parseInt(path.pop(), 10);
```

Comment:
This will unharden the `account` which I believe is the expected behavior. I think a comment in the jsdoc could be helpful here

```javascript
> parseInt("0'", 10)
0
```

Reply:
You are correct; the behavior is expected. I can add a comment about this object assuming hardened derivation above the change level.

Reply:
maybe move path logic to `fromPath` (because there's `toPath`)

Reply:
@nodar-chkuaselidze I think the `toPath` method is poorly named. The name suggests the method _creates_ a `Path` object, but it _actually_ creates a String representation of the derivation path.

To me, the name `fromPath` suggests it would take a `Path` object and return an instance of some other object, but given the context, it would need to return an instance of Path (or at least a hash with attributes used to construct a Path object).

This ambiguity makes me lean towards keeping the deserialization logic for the `derivation` attribute here.

Reply:
> I think the toPath method is poorly named.

I agree that it sounds like `toPath` would return an instance of the `Path` class.
What do you think of using `pop` vs `path[path.length - 1]`? I like the latter more personally because its not mutating the `path` array.

I think that its a good idea to assert that `path.length >= 3` because we don't want any of `account`, `branch` or `index` to be `undefined`.

Reply:
@tynes I do not assert the length of `path` explicitly, but `account`, `branch`, and `index` are asserted to match the values passed from the JSON object. That said, I'll add assertions for the proper lengths of 4 and 6 (the "m" will be present in the path array). Any other length will cause an assertion error.

Do you mind elaborating on your concerns regarding mutating the path array?

Reply:
> Do you mind elaborating on your concerns regarding mutating the path array?

Its just an additional mental overhead for future people that would like to refactor the code. Not the worst, especially in this case since its a pretty simple codepath.

Reply:
I'll add a comment that notes the path object is mutated.